### PR TITLE
Workflow

### PR DESCRIPTION
**Pull request by martemoslet**

### CODE REVIEW

File: .github/workflows/static.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (Fermain) [inline .github/workflows/static.yml:32]: You should install and build the project too as extra steps here

File: cypress/e2e/login.cy.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    })
+
+
+    it("user cannot submit login form with invalid credentials", () => {
+        cy.get("#loginForm").should("be.visible");
+        cy.get("#loginEmail").should("be.visible").type(`invalid@email.no`).then(($email) => {
+            expect($email[0].validationMessage).to.not.be.empty
+
+        cy.get("#loginPassword").should("exist").type(`passord{enter}`)
+
```
Comment (Fermain) [inline cypress/e2e/login.cy.js]: I'm not sure about this test. I haven't run it yet myself just looking at the code it seems like you should enter the password first, then include {enter} when typing the email field?

File: src/js/api/auth/logout.test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        }),
+        status: 200,
+        statusText: "OK",
+        ok: true
+    })
+);
+
+
+describe("logout", () => {
+    localStorage.setItem(profile, accessToken);
```
Comment (Fermain) [inline src/js/api/auth/logout.test.js]: Please run a formatter on jest files to fix their indentation

File: src/js/api/posts/create.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    ok: true,
+    json: jest.fn().mockResolvedValue(mockpost)
+});
+
+global.fetch = mockfetch;
+
+describe("create", () => {
+    it("returns a valid item with a valid input", async () => {
+        const post = await createPost("title", ["tag1", "tag2"], "", "body");
+        expect(post).toMatchObject(mockpost);
```
Comment (Fermain) [inline src/js/api/posts/create.test.js]: You could also `expect(fetch).toHaveBeenCalled()` or similar

File: src/js/storage/save.test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { save } from "./save.js";
+import storage from "../helpers/storage";
+import { Exception } from "sass";
+
+const testValue = { test: true };
+
+global.localStorage = storage;
+
+describe("save", () => {
+    it("should save the correct value to localStorage", () => {
```
Comment (Fermain) [inline src/js/storage/save.test.js]: It would be nice to see this added to the logout test too

File: .eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": "eslint:recommended",
+    "overrides": [
+        {
+            "files": ["**/*.test.js"],
```
Comment (Fermain) [inline .eslintrc.json]: I think you need to add your helper file here too

**Review by Fermain — CHANGES_REQUESTED** [review]
There are some minor issues to be resolved but overall the quality of this PR is good. Tests are not all robust yet but the range of syntax and approaches is wide and there is clear engagement with the difficult concepts underlying testing.

I hope that I have made it clear what is an optional change and what is required, you would most likely pass with what you have here already but with a little extra work it can be elevated to a strong submission.

File: .github/workflows/static.yml (hunk 1 above)
Comment (martemoslet) [reply .github/workflows/static.yml:32]: Ok, I have made this change now

File: cypress/e2e/login.cy.js (hunk 2 above)
Comment (martemoslet) [reply cypress/e2e/login.cy.js]: I'm not sure what you mean here, do you mean that the expect should be after entering password and clicking enter?

Comment (Fermain) [reply cypress/e2e/login.cy.js]: No I mean that you should type the password first, then type the email then hit enter

Comment (martemoslet) [reply cypress/e2e/login.cy.js]: Thanks, got it. That works, and I have made that change

File: src/js/api/auth/logout.test.js (hunk 3 above)
Comment (martemoslet) [reply src/js/api/auth/logout.test.js]: Do you mean running prettier or lint-staged over those files?

File: .eslintrc.json (hunk 6 above)
Comment (martemoslet) [reply .eslintrc.json]: Ok, that is done

File: src/js/storage/save.test.js (hunk 5 above)
Comment (martemoslet) [reply src/js/storage/save.test.js]: Do you mean a new file like remove.test.js ? Or something like that in the logout test?

Comment (Fermain) [reply src/js/storage/save.test.js]: I mean this approach could be used to test logout

File: src/js/api/auth/logout.test.js (hunk 3 above)
Comment (Fermain) [reply src/js/api/auth/logout.test.js]: Yeah or the built in formatter in vscode

Comment (martemoslet) [reply src/js/api/auth/logout.test.js]: Ok, thanks :) That is done

File: src/js/storage/save.test.js (hunk 5 above)
Comment (martemoslet) [reply src/js/storage/save.test.js]: Ok, I will try

File: src/js/api/posts/create.test.js (hunk 4 above)
Comment (martemoslet) [reply src/js/api/posts/create.test.js]: I have done some changes to this test now, not sure if it's exactly what you meant though

File: src/js/storage/save.test.js (hunk 5 above)
Comment (martemoslet) [reply src/js/storage/save.test.js]: I have updated this test now

**Review by Fermain — APPROVED** [review]
Thanks for implementing the feedback

**Pull request merged**